src/client/canvas/svg/svg.tsx

```diff
@@ -0,0 +1,82 @@
+import { Component } from 'react'
+import * as React from 'react'
+
+import { Transform } from '../../math/transform'
+import { Appearance } from '../appearance/appearance'
+import { BasicAppearance } from '../appearance/basic_appearance'
+import { LineAppearance } from '../appearance/line_appearance'
+import { ArcGeometry } from '../geometry/arc_geometry'
+import { ArrowGeometry } from '../geometry/arrow_geometry'
+import { CircleGeometry } from '../geometry/circle_geometry'
+import { LineGeometry } from '../geometry/line_geometry'
+import { MarkerGeometry } from '../geometry/marker_geometry'
+import { PolygonGeometry } from '../geometry/polygon_geometry'
+import { TextGeometry } from '../geometry/text_geometry'
+import { Group as GroupGeometry } from '../object/group'
+import { Object2d } from '../object/object2d'
+import { Geometry } from '../object/shape'
+import { Shape } from '../object/shape'
+
+import { Arc } from './arc'
+import { Arrow } from './arrow'
+import { Circle } from './circle'
+import { Group } from './group'
+import { Line } from './line'
+import { Marker } from './marker'
+import { Polygon } from './polygon'
+import { Text } from './text'
+
+export function appearance(appearance: Appearance): {} {
+  if (appearance instanceof BasicAppearance) {
+    return {
+      fill: appearance.fillStyle,
+      strokeWidth: appearance.lineWidth,
+      stroke: appearance.strokeStyle,
+    }
+  } else if (appearance instanceof LineAppearance) {
+    return {
+      strokeLinecap: appearance.lineCap,
+      strokeLinejoin: appearance.lineJoin,
+      strokeDashoffset: appearance.lineDashOffset,
+      strokeWidth: appearance.lineWidth,
+      stroke: appearance.strokeStyle,
+    }
+  } else {
+    throw new Error(`Unsupported appearance type ${appearance}`)
+  }
+}
+
+export function transform(transform: Transform): string {
```

Comment:
`transform` is a very common variable name and will conflict with the rest of my suggestions, can these two above functions be named `svgTransform` and `svgApperance`?

Reply:
👍